bom/application/pom.xml

```diff
@@ -881,6 +881,26 @@
                 <artifactId>quarkus-keycloak-admin-client-deployment</artifactId>
                 <version>${project.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-keycloak-admin-client-cdi</artifactId>
```

Comment:
What were your reasons to make it a separate extension? CDI is usually at the core of all our extensions and we don't have a separate one to bring CDI support.

Note that there might be good reasons but we need to carefully discuss this before thinking about merging this.

Reply:
@gsmet there is classic and reactive Keycloak admin client and we want (I did :-)) to use one CDI on both of them.

Reply:
Please note that I added unlisted flag to true, so in my understanding the CDI won't be listed to users...

Reply:
Hi @michalvavrik Thanks for the PR, I'll have a deeper look next week, but I did not expect that 2 new modules would have to be added too :-), I'd just have it all bundled in with the existing extension, there is not much there to add, the config group, and provding a synthetic bean, which s what we do for OIDC, and everywhere else when the injection support is needed for non real beans. Think about it please, thanks

Reply:
Alright, as I've time before you review this, I'll try to think about it. 

Just to be honest, I'm not sure if you guys thought about it (might be wrong!), because following classes would have to be duplicated (f.e. you can't bring `quarkus-core-deployment` to runtime deps and so on): `io.quarkus.keycloak.adminclient.cdi.deployment.KeycloakAdminClientCDIProcessor`, `io.quarkus.keycloak.adminclient.cdi.runtime.KeycloakAdminClientRecorder`, and build items....

I could extract business logic to an abstraction level (one new module) and duplicate build steps and recorder that would merely call the abstraction level....

Reply:
That would be better than having user-facing artifacts. Better hide things from the user. If you have a look at some other extensions, we are using this very pattern (with some `-common` extensions).

The important point is that CDI is not optional in Quarkus: it's at its very core. Thus our extensions are CDI-enabled by default.

Reply:
CDI is now integral part of keycloak-admin-client/reactive, the new modules can't be avoided:
- `common` for config and validation
- `SPI` for admin client <-> oidc

I checked existing extensions several times and can't see significant the difference in current approach, but I'm happy to learn/adjust if you do! Thank you